src/Simplex/Chat.hs

```diff
@@ -4934,6 +4979,21 @@ processAgentMessageConn user@User {userId} corrId agentConnId agentMessage = do
     checkHostRole GroupMember {memberRole, localDisplayName} memRole =
       when (memberRole < GRAdmin || memberRole < memRole) $ throwChatError (CEGroupContactRole localDisplayName)
 
+    xGrpMemCon :: GroupInfo -> GroupMember -> MemberId -> m ()
+    xGrpMemCon gInfo m memId = do
+      refMember <- withStore $ \db -> getGroupMemberByMemberId db user gInfo memId
```

Comment:
reuse `getGroupMemberByMemberId` in places where all members are read unnecessarily (search "getGroupMembers")